### PR TITLE
SDCICD-591: skip certain cluster checks in e2e when using the mock provider

### PR DESCRIPTION
**Pull request by jeefy**
This allows addon authors the ability to use/specify a Kubeconfig externally and test against any valid openshift cluster while using the `mock` provider. This isn't an ideal solution and, in the future, we need to create a "local" provider that expects a Kubeconfig and lets us remove all these conditional statements within the main e2e code. :)

### CODE REVIEW

File: pkg/e2e/e2e.go [hunk 1]
```diff
@@ -105,7 +105,7 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 		viper.Set(config.CloudProvider.Region, cluster.Region())
 		log.Printf("CLOUD_PROVIDER_REGION set to %s from OCM.", viper.GetString(config.CloudProvider.Region))
 
-		if !viper.GetBool(config.Addons.SkipAddonList) {
+		if !viper.GetBool(config.Addons.SkipAddonList) || viper.GetString(config.Provider) == "mock" {
```
Comment (whereswaldon) [inline pkg/e2e/e2e.go:108]: Why do this when the provider *is* the mock provider? It seems to me that you'd want the opposite

**Review by whereswaldon — CHANGES_REQUESTED** [review]

File: pkg/e2e/e2e.go (hunk 1 above)
Comment (jeefy) [reply pkg/e2e/e2e.go:108]: You are _so_ so right.

Comment (jeefy) [reply pkg/e2e/e2e.go]: Fixed! PTAL

**Review by whereswaldon — APPROVED** [review]

**Pull request merged**